File: backend/models/entry.py
from pydantic import BaseModel, computed_field
from typing import Optional

class Entry(BaseModel):
    """Pydantic model for a Nightscout entry."""
    _id: str
    type: str
    
    date: int
    dateString: str
    cached_at: str
    sgv: Optional[int] = None
    trend: int
    direction: str
    device: str
    utcOffset: int
    sysTime: str
    # mills: int
    
    @computed_field
    @property
    def mmol(self) -> float:
        """Return sgv converted to mmol/L (sgv / 18), rounded to one decimal."""
        if self.sgv is not None:
            return round(self.sgv / 18, 1)
        return None

    

File: frontend/src/models/entry.ts
export interface Entry {
  _id: string;
  type: string;
  date: number;
  dateString: string;
  date_date: string;
  cached_at: string;
  sgv?: number;
  trend?: number;
  direction?: string;
  device: string;
  utcOffset: number;
  sysTime: string;
  mmol: number;
}
